### ISSUE TITLE
Fix missing texture race condition

### ISSUE THREAD

**Issue by xJon** [Kanzaji/XU2-Patcher#1]
See https://github.com/rwtema/extrautilities_old_issues/issues/1812, https://github.com/rwtema/Extra-Utilities-2-Source/issues/178, https://github.com/rwtema/Extra-Utilities-2-Source/issues/436, https://github.com/Krutoy242/Enigmatica2Expert-Extended/pull/146.

Recently https://github.com/xJon/The-1.12.2-Pack/issues/973.

Would be great to have a proper fix!

**Comment by Kanzaji** [comment]
Gonna take a look at this in my free time asap 😄 Thanks for the patch request.

**Comment by Kanzaji** [comment]
Okay this might be harder than it looked... it just worked in Dev >.>

![20240904_142316.jpg](https://github.com/user-attachments/assets/f7c8e50e-d4ab-4df6-b2dd-23586d132076)

PS: Ignore phone photo - i was testing this at school lmao.

**Comment by xJon** [comment]
Indeed, that is why I referred to it as a race condition - perhaps you can try The 1.12.2 Pack to see the broken textures in action, although results can vary within the same modpack using different hardware.

**Comment by Kanzaji** [comment]
> Indeed, that is why I referred to it as a race condition - perhaps you can try The 1.12.2 Pack to see the broken textures in action, although results can vary within the same modpack using different hardware.

The biggest issue with this (and most likely why Thema didn't touch this) is that I think I don't even have a way to debug this >.> So unless I somehow figure out why this is happening, it might be a really hard fix.

Not that I'm giving up on it, but I don't have time to look into this in the week ^^"

**Comment by xJon** [comment]
No worries - it is not urgent. What about simply changing the way the textures are applied?

**Comment by xJon** [comment]
Bumping this just in case it was forgotten 🙃

**Comment by Kanzaji** [comment]
No worries it wasn't 😄 I just didn't had a lot of time to look into this recently >.>

**Comment by Kanzaji** [comment]
https://github.com/3TUSK/RailwayDepotWorker/issues/10
